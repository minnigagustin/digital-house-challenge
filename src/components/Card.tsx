import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { scale, verticalScale } from "utils/index";

interface Props {
  points: number;
  month: string;
}

export function Card({ month, points }: Props) {
  return (
    <View style={styles.container}>
      <Text style={styles.date}>{month}</Text>
      <Text style={styles.points}>{points} pts</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center",
    paddingTop: verticalScale(8),
    backgroundColor: "#334FFA",
    height: verticalScale(120),
    width: "80%",
    borderRadius: scale(20),
    paddingHorizontal: scale(19),
    shadowColor: "black",
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,
    elevation: 5,
  },
  date: {
    position: "absolute",
    left: scale(19),
    top: verticalScale(21),
    color: "white",
    fontSize: 16,
    fontWeight: "800",
    textTransform: "capitalize",
    fontFamily: "AvenirBold",
  },
  points: {
    color: "white",
    fontWeight: "bold",
    fontSize: scale(32),
    fontFamily: "AvenirBold",
  },
});
