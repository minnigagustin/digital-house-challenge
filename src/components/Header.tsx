import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { scale, verticalScale } from "utils/index";

interface Props {
  name: string;
}

export function Header({ name }: Props) {
  return (
    <View>
      <Text style={styles.welcome}>Bienvenido de vuelta!</Text>
      <Text style={styles.userName}>{name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  welcome: {
    fontSize: scale(16),
    fontWeight: "800",
    fontFamily: "AvenirBold",
  },
  userName: {
    fontSize: scale(14),
    fontFamily: "AvenirRegular",
    lineHeight: verticalScale(22),
  },
});
