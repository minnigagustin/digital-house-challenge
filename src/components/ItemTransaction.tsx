import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { Image } from "expo-image";
import React from "react";
import { Text, View, StyleSheet, Pressable } from "react-native";
import { formatDate } from "utils/index";

export interface Props {
  date: Date;
  product: string;
  points: number;
  image: string;
  is_redemption: boolean;
  id: string;
}

export function ItemTransaction({
  image,
  product,
  date,
  is_redemption,
  points,
}: Props) {
  const { navigate } = useNavigation<NativeStackNavigationProp<any>>();

  const handlePress = () =>
    navigate("ProductDetails", {
      date,
      image,
      points,
      product,
    });

  return (
    <Pressable onPress={handlePress}>
      <View style={styles.container}>
        <>
          <Image
            style={styles.image}
            source={{ uri: image }}
            priority={"high"}
          />
          <View style={styles.infoContainer}>
            <Text style={styles.product}>{product}</Text>
            <Text style={styles.date}>{formatDate(date)}</Text>
          </View>
        </>
        <>
          {is_redemption ? (
            <Text style={styles.min}>
              -<Text style={styles.points}>{`${points}  >`}</Text>
            </Text>
          ) : (
            <Text style={styles.sum}>
              +<Text style={styles.points}>{`${points}  >`}</Text>
            </Text>
          )}
        </>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    padding: 10,
  },
  image: {
    width: 55,
    height: 55,
    marginRight: 11,
    borderRadius: 10,
  },
  infoContainer: {
    flex: 1,
  },
  product: {
    fontSize: 14,
    fontWeight: "800",
    color: "#000",
    fontFamily: "AvenirBold",
  },
  date: {
    fontSize: 12,
    fontWeight: "400",
    marginTop: 7,
    color: "#000",
    fontFamily: "AvenirMedium",
  },
  min: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#FF0000",
  },
  sum: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#00B833",
  },
  points: {
    fontFamily: "AvenirBold",
    fontSize: 16,
    fontWeight: "800",
    color: "#000",
  },
});
