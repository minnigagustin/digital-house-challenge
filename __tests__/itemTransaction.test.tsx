import { ItemTransaction } from "components/ItemTransaction";
import React from "react";
import renderer from "react-test-renderer";

const date = new Date("2022-12-09T10:20:00.909Z");
const product = "Fantastic Granite Bacon";
const points = 42416;
const image = "https://loremflickr.com/640/480/technics";
const is_redemption = false;
const id = "3";

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

describe("itemTransaction", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <ItemTransaction
          date={date}
          product={product}
          points={points}
          image={image}
          is_redemption={is_redemption}
          id={id}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
