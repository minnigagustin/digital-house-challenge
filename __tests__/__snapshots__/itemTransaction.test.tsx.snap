// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemTransaction should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      borderRadius={10}
      height={55}
      marginRight={11}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      proxiedProperties={
        {
          "contentFit": "cover",
          "contentPosition": {
            "left": "50%",
            "top": "50%",
          },
          "placeholder": [],
          "priority": "high",
          "source": [
            {
              "uri": "https://loremflickr.com/640/480/technics",
            },
          ],
          "tintColor": undefined,
          "transition": null,
        }
      }
      style={
        {
          "borderRadius": 10,
          "height": 55,
          "marginRight": 11,
          "width": 55,
        }
      }
      width={55}
    />
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "AvenirBold",
            "fontSize": 14,
            "fontWeight": "800",
          }
        }
      >
        Fantastic Granite Bacon
      </Text>
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "AvenirMedium",
            "fontSize": 12,
            "fontWeight": "400",
            "marginTop": 7,
          }
        }
      >
        9 de diciembre, 2022
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#00B833",
          "fontFamily": "AvenirBold",
          "fontSize": 16,
          "fontWeight": "800",
        }
      }
    >
      +
      <Text
        style={
          {
            "color": "#000",
            "fontFamily": "AvenirBold",
            "fontSize": 16,
            "fontWeight": "800",
          }
        }
      >
        42416  &gt;
      </Text>
    </Text>
  </View>
</View>
`;
